src/net/read_url.rs

```diff
@@ -3,6 +3,7 @@ use anyhow::{anyhow, format_err};
 use crate::context::Context;
 use crate::socks::Socks5Config;
 
+/// Retrieves the contents of URL using HTTP GET request.
 pub async fn read_url(context: &Context, url: &str) -> anyhow::Result<String> {
```

Comment:
is the result always a String? doesn't `String` do some unicode checks that might fail when loading an image for example?